cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -552,8 +523,9 @@ func (m *McmManager) resetPriorityForMachines(mcRefs []*Ref) error {
 }
 
 // prioritizeMachinesForDeletion prioritizes the targeted machines by updating their priority annotation to 1
-func (m *McmManager) prioritizeMachinesForDeletion(targetMachineRefs []*Ref) (int, error) {
+func (m *McmManager) prioritizeMachinesForDeletion(targetMachineRefs []*types.NamespacedName) ([]string, error) {
 	var expectedToTerminateMachineNodePairs = make(map[string]string)
+	var machinesMarkedWithPrio1 []string
```

Comment:
`machinesMarkedWithPrio1` -> `prio1MarkedMachineNames`. Also move this variable to a named return value on this method.

Reply:
maps.Keys was introduced in go 1.23 - https://pkg.go.dev/maps@go1.22.10 does not have Keys function

Reply:
Hmm...my bad this method was added later to the package.